firmware/application/apps/ble_app.cpp

```diff
@@ -0,0 +1,403 @@
+    line += ":" + to_string_hex((entry.macAddress >> 24) & 0xFF, 2);
+    line += ":" + to_string_hex((entry.macAddress >> 32) & 0xFF, 2);
+    line += ":" + to_string_hex((entry.macAddress >> 40), 2);
+
+    // Handle spacing for negative sign.
+    uint8_t db_spacing = entry.dbValue > 0 ? 7 : 6;
+
+    // Pushing single digit values down right justified.
+    if (entry.dbValue > 9 || entry.dbValue < -9) {
+        db_spacing--;
+    }
+
+    line += pad_string_with_spaces(db_spacing) + to_string_dec_int(entry.dbValue);
+
+    line.resize(target_rect.width() / 8, ' ');
+    painter.draw_string(target_rect.location(), style, line);
+}
+
+BleRecentEntryDetailView::BleRecentEntryDetailView(NavigationView& nav, const BleRecentEntry& entry)
+    : nav_{nav},
+      entry_{entry} {
+    add_children({&button_done,
+                  &labels});
+
+    button_done.on_select = [this](const ui::Button&) {
+        nav_.pop();
+    };
+}
+
+void BleRecentEntryDetailView::update_data() {
+}
+
+void BleRecentEntryDetailView::focus() {
+    button_done.focus();
+}
+
+Rect BleRecentEntryDetailView::draw_field(
+    Painter& painter,
+    const Rect& draw_rect,
+    const Style& style,
+    const std::string& label,
+    const std::string& value) {
+    const int label_length_max = 4;
+
+    painter.draw_string(Point{draw_rect.left(), draw_rect.top()}, style, label);
+    painter.draw_string(Point{draw_rect.left() + (label_length_max + 1) * 8, draw_rect.top()}, style, value);
+
+    return {draw_rect.left(), draw_rect.top() + draw_rect.height(), draw_rect.width(), draw_rect.height()};
+}
+
+void BleRecentEntryDetailView::paint(Painter& painter) {
+    View::paint(painter);
+
+    const auto s = style();
+    const auto rect = screen_rect();
+
+    auto field_rect = Rect{rect.left(), rect.top() + 16, rect.width(), 16};
+
+    uint8_t type[5];
+    uint8_t length[5];
+    uint8_t data[5][40];
+
+    int currentByte = 0;
+    int currentPacket = 0;
+    int i = 0;
+    int j = 0;
+    int k = 0;
+
+    for (currentByte = 0; (currentByte < entry_.packetData.dataLen) && (currentPacket < 5);) {
+        length[currentPacket] = entry_.packetData.data[currentByte++];
+        type[currentPacket] = entry_.packetData.data[currentByte++];
+
+        // This should never happen, but in here just in case.
+        // Break because we can't trust rest of data.
+        // if (length[currentPacket] > entry_.packetData.dataLen)
+        // {
+        //     break;
+        // }
+
+        // Subtract 1 because type is part of the length.
+        for (i = 0; i < length[currentPacket] - 1; i++) {
+            data[currentPacket][i] = entry_.packetData.data[currentByte++];
+        }
+
+        currentPacket++;
+    }
+
+    for (i = 0; i < currentPacket; i++) {
+        uint8_t number_data_lines = ceil((float)(length[i] - 1) / 10.0);
+        uint8_t current_line = 0;
+        std::array<std::string, 5> data_strings{{""}};
```

Comment:
You don't need to initialize the strings here, the string constructor will be run for you.
`std::array<std::string, 5> data_strings{}` would be fine.